tools/SourceKit/tools/sourcekitd/lib/API/sourcekitdAPI-XPC.cpp

```diff
@@ -247,7 +247,7 @@ void ResponseBuilder::Dictionary::setCustomBuffer(
   std::unique_ptr<llvm::MemoryBuffer> CustomBuf;
   CustomBuf = llvm::WritableMemoryBuffer::getNewUninitMemBuffer(
       sizeof(uint64_t) + MemBuf->getBufferSize());
-  char *BufPtr = (char*)CustomBuf->getBufferStart();
+  char *BufPtr = const_cast<char*>(CustomBuf->getBufferStart());
```

Comment:
I think the real problem is that the `unique_ptr` owns a `llvm::MemoryBuffer` instead of a `llvm::WritableMemoryBuffer` (which does the `const_cast` for you, the evil thing).